### ISSUE TITLE
Duplicate Shader Key Files

### ISSUE THREAD

**Issue by tebjan** [stride3d/stride#32]
after upgrading from xenko 2.1 all custom shaders in visual studio 2017 have duplicate key files. at least game studio thinks so, although they are not actually present on disk. is there any order of things i can do to fix that? or how would i upgrade my project that this doesn't happen? i could reset the git repo and start the upgrade process again, but in which order should i open/upgrade/save the project files in game studio and visual studio?

EDIT: i have to add, the project was a VS 2015 solution. the errors occurred with 2017. currently trying to get the 2.1 version of the project running with VS 2017 and then i'll try another upgrade.

these are the errors:
![unbenannt](https://user-images.githubusercontent.com/1094716/43674764-ae421234-97d8-11e8-9c60-796e9dd84726.PNG)

then on build:
![unbenannt2](https://user-images.githubusercontent.com/1094716/43674766-b8187014-97d8-11e8-8496-1882ea2ae9f6.PNG)

**Comment by tebjan** [comment]
might be related to this issue on the old repo: https://github.com/SiliconStudio/xenko/issues/672

**Comment by tebjan** [comment]
might this be a problem of the project upgrader as described here #33 ?
how should i re-include the shader files in order to get them properly handled by VS and GS?

**Comment by xen2** [comment]
Yes it is related to https://github.com/xenko3d/xenko/issues/33#issuecomment-410451335
Sorry about that, it's a one time thing that shouldn't happen during next update due to the csproj changes.
Please open your project in Visual Studio, include the .xksl file as part of your project. Provided you installed the Xenko VS Plugin, it should automatically mark those files with the Custom Tool `XenkoShaderKeyGenerator` and the .cs embedded as a generated item.

![image](https://user-images.githubusercontent.com/527565/43677647-5d81572a-9840-11e8-95b0-cf1a3573c973.png)

If you have any trouble with Visual Studio, the .csproj looks like this:
```
  <ItemGroup>
    <Compile Update="Test.cs">
      <DesignTime>True</DesignTime>
      <AutoGen>True</AutoGen>
      <DependentUpon>Test.xksl</DependentUpon>
    </Compile>
    <None Update="Test.xksl">
      <Generator>XenkoShaderKeyGenerator</Generator>
      <LastGenOutput>Test.cs</LastGenOutput>
    </None>
  </ItemGroup>
```
